bpf/lib/nodeport.h

```diff
@@ -1777,10 +1789,17 @@ static __always_inline int nodeport_lb4(struct __ctx_buff *ctx,
 		if (nodeport_uses_dsr4(&tuple))
 			return CTX_ACT_OK;
 #endif
-
 		ctx_store_meta(ctx, CB_NAT, NAT_DIR_INGRESS);
 		ctx_store_meta(ctx, CB_SRC_IDENTITY, src_identity);
-		ep_tail_call(ctx, CILIUM_CALL_IPV4_NODEPORT_NAT);
+		lb4_populate_ports(ctx, &tuple, l4_off);
```

Comment:
I am bit missing how in the case of NAT64 we can enter this code path. In any case, a comment would be useful.

Reply:
NAT64 basically transforms a v6 req into a v4 req, so the LB to backend path is v4, and then this will be his via reply from backend. Will add a comment.